src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -13,15 +15,40 @@ internal partial class DeclarationNameCompletionProvider
     {
         internal class NameGenerator
         {
-            internal static ImmutableArray<Words> GetBaseNames(ITypeSymbol type)
+            internal static ImmutableArray<Words> GetBaseNames(ITypeSymbol type, bool pluralize)
             {
                 var baseName = TryRemoveInterfacePrefix(type);
                 var parts = StringBreaker.GetWordParts(baseName);
                 var result = GetInterleavedPatterns(parts, baseName);
+                if (pluralize)
```

Comment:
it feels like pluralize should be passed to GetInterleavedPatterns so it can make the right words in place.